### PR TITLE
Added BSD Licence to all checks.

### PR DESCRIPTION
**Pull request by BenStarmerSmith**
## Proposed changes

Added BSD Licence to all checks.

## Checklist

Before posting a PR ensure that from each of the below categories **AT LEAST ONE BOX HAS BEEN CHECKED**. If more than one category is applicable then more can be checked. Also ensure that the proposed changes have been filled out with relevant information for reviewers.

## Tests

- [ ] No tests required to be added. (For small changes that will be tested by CI/CD infrastructure).
- [x] Added/Modified automated and PhantomHand CI tests (if a new class is added (Python or C++), the interface of that class must be unit tested).
- [ ] Manually tested in simulation (if simulation specific or no hardware required to test the functionality). 
- [ ] Manually tested on hardware (if hardware specific or related).

## Documentation

- [ ] No documentation required to be added.
- [ ] Added documentation (For any new feature, explain what it does and how to use it. Write the documentation in a relevant space, e.g. Github, Confluence, etc).
- [ ] Updated documentation (For changes to pre-existing features mentioned in the documentation).

### CODE REVIEW

**Comment by BenStarmerSmith** [comment]
Screenshot of missing files:
Screenshot from 2022-08-26 15-57-25

**Comment by BenStarmerSmith** [comment]
Files missing current year:
Screenshot from 2022-08-26 15-58-49

File: ansible/roles/ci/code_style_check/license/files/gather_missing_licences.py [hunk 1]
```diff
@@ -118,19 +120,29 @@ def gather_missing_licences(data):
...
                 line = line.strip()  # Remove whitespaces so we can find lines with just comments
-                if extension == "py":
+                if extension in ["py", "msg", "yml", "yaml", "sh"]:
                     if line and len(line) > 1 and line[0] == "#":
                         if "Shadow Robot Company Ltd" in line and "Copyright" in line:
                             if data.current_year not in line:
                                 missing_licence.append(file_path)
-                else:
+                elif extension in ["c", "cpp", "h", "hpp"]:
                     if line and len(line) > 1 and line[0] in ["/","*"]:
```
Comment (maxzieba) [inline ansible/roles/ci/code_style_check/license/files/gather_missing_licences.py:133]: can't this be packed into one if **if line and len(line) > 1 and line[0] in ["/","*", "#"]** as the inside seems to be the same?

File: ansible/roles/ci/code_style_check/license/files/gather_missing_licences.py [hunk 2]
```diff
@@ -14,14 +14,16 @@
...
 import sys
 import argparse
 import subprocess
 from datetime import date
 
-PYTHON_HEADERS = ["#!/usr/bin/env python", "#!/usr/bin/python"]
-ACCEPTED_EXTENSIONS = ["py", "c", "h", "cpp", "hpp"]
+# Include bash in python files as we also put licences in bash files too.
+PYTHON_HEADERS = ["#!/usr/bin/env python", "#!/usr/bin/python", "#!/bin/bash", "#!/usr/bin/env bash"]
+ACCEPTED_EXTENSIONS = ["py", "c", "h", "cpp", "hpp", "msg", "yml", "yaml", "sh", "xml", "xacro", "dae", "launch"]
```
Comment (maxzieba) [inline ansible/roles/ci/code_style_check/license/files/gather_missing_licences.py:26]: I assume that msg extension is for ROS messages? Shouldn't we also have then .srv and .action added?

File: ansible/roles/ci/code_style_check/license/files/gather_missing_licences.py (hunk 1 above)
Comment (BenStarmerSmith) [reply ansible/roles/ci/code_style_check/license/files/gather_missing_licences.py:133]: Yes good point!

File: ansible/roles/ci/code_style_check/license/files/gather_missing_licences.py (hunk 2 above)
Comment (BenStarmerSmith) [reply ansible/roles/ci/code_style_check/license/files/gather_missing_licences.py:26]: I guess thats true, I just based this off the documentation I got :) do .srv and .action use # based comments?

Comment (maxzieba) [reply ansible/roles/ci/code_style_check/license/files/gather_missing_licences.py:26]: I would say its the same 'type' as .msg so if we are adding licenses in the messages then the same should happen in .srv and .action

**Review by maxzieba — APPROVED** [review]

**Review by bdenoun — APPROVED** [review]

**Pull request merged**